pydm/display.py

```diff
@@ -67,6 +72,28 @@ def load_file(file, macros=None, args=None, target=ScreenTarget.NEW_PROCESS):
     return w
 
 
+@lru_cache
+def _compile_ui_file(uifile: str) -> Tuple[str, str]:
+    """
+    Compile the ui file using uic and return the result as a string along with the associated class name.
+    Caches the result to improve performance when the same ui file is re-used many times within a display.
+
+    Parameters
+    ----------
+    uifile : str
+        The path to a .ui file to compile
+
+    Returns
+    -------
+    Tuple[str, str] - The first element is the compiled ui file, the second is the name of the class (e.g. Ui_Form)
+    """
+    code_string = StringIO()
+    uic.compileUi(uifile, code_string)
+    # Grabs non-whitespace characters between class and the opening parenthesis
+    class_name = re.search(r'class\s*(\S*)\(', code_string.getvalue()).group(1)
```

Comment:
Matching at the beginning of the line might be a good idea, even if not strictly required with the current uic-generated code:
```suggestion
    class_name = re.search(r'^class\s*(\S*)\(', code_string.getvalue()).group(1)
```

Reply:
Good idea, thanks! This does make it more robust, I'll need to stick a `re.MULTILINE` parameter in there as well to get past the auto-generated documentation and PyQt5 import statements.
`re.search(r'^class\s*(\S*)\(', code_string.getvalue(), re.MULTILINE).group(1)`